core/src/utils/framework-delegate.ts

```diff
@@ -86,7 +86,7 @@ export const CoreDelegate = () => {
       BaseComponent.appendChild(el);
 
       await new Promise((resolve) => componentOnReady(el, resolve));
-    } else if (BaseComponent.children.length > 0) {
+    } else if (BaseComponent.children.length > 0 && cssClasses.length > 0) {
```

Comment:
This change is intentional. Without it, the logic of cloning to a parent element causes exceptions and broken behavior when the loading component is used inline. 

The logic here is only intended to reflect specified CSS classes to a host element. When there is no CSS classes, we can skip that behavior.

Reply:
I agree that we should skip this, but I don't think this is the correct check as it will break modals/popovers without css classes. This if-block is designed to ensure that user-specified components passed into `ion-modal` and `ion-popover` have a single single parent element.

Example:

```html
<ion-modal>
  <my-cmp></my-cmp>
  <my-cmp></my-cmp>
</ion-modal>
```

would become

```html
<ion-modal>
  <div class="ion-page">
    <my-cmp></my-cmp>
    <my-cmp></my-cmp>
  </div>
</ion-modal>
```

For `ion-loading` this is not needed because there is no user-specified components. Instead, this code was incorrectly moving the internals of `ion-loading`.

Perhaps we could add add some kind of `hasUserDefinedComponent` check? It could check to see if `BaseComponent.children.length > 0` and if the component is a modal or popover.